src/otel/src/lib.rs

```diff
@@ -0,0 +1,319 @@
+use opentelemetry::global::BoxedSpan;
+use opentelemetry::trace::{
+    Span, SpanBuilder, SpanKind, TraceId, Tracer, TracerProvider,
+};
+use opentelemetry::{global, KeyValue};
+use opentelemetry_otlp::Protocol;
+use opentelemetry_otlp::WithExportConfig;
+use opentelemetry_sdk::trace::{Config, BatchConfigBuilder, Sampler};
+use opentelemetry_sdk::{runtime, Resource};
+use std::ffi::{CStr, CString};
+use std::{ptr, time};
+use std::ptr::addr_of;
+use std::slice;
+use std::sync::{Arc, OnceLock};
+use tokio::sync::mpsc;
+use tokio::sync::mpsc::{Receiver, Sender};
+
+
+const TRACEPARENT_HEADER_LEN: u8 = 55;
+const TIMEOUT: time::Duration = std::time::Duration::from_secs(10);
+
+#[repr(C)]
+pub struct nxt_str_t {
+    pub len: usize,
```

Comment:
Could we make this `length` please?,  so it matches the C code (and wasm-wasi-component).